synapse/storage/databases/main/account_data.py

```diff
@@ -546,6 +546,45 @@ def _add_account_data_for_user(
         for ignored_user_id in previously_ignored_users ^ currently_ignored_users:
             self._invalidate_cache_and_stream(txn, self.ignored_by, (ignored_user_id,))
 
+    async def purge_account_data_for_user(self, user_id: str) -> None:
+        """
+        Removes ALL the account data for a user.
+        Intended to be used upon user deactivation.
+
+        Also purges the user from the ignored_users cache table
+        and the push_rules cache tables.
```

Comment:
Are these part of account data or are they unrelated? (This is confusing to me that they're called out separately.)

Reply:
They're caches of the information in the account data — much like we have `state_events` to cache state events even though the canonical information is in `event_json`.
Will clarify since I suppose it's non-obvious.

Reply:
I wasn't really sure what "cache table" meant, I suppose. Would it make more sense to just list all the tables it clears? Or call them derived or something?